man/kernel-install.xml

```diff
@@ -167,6 +173,10 @@
 
     <para><varname>KERNEL_INSTALL_MACHINE_ID=</varname> is set for the plugins to the desired <replaceable>MACHINE-ID</replaceable>.</para>
     <para><varname>KERNEL_INSTALL_BOOT_ROOT=</varname> is set for the plugins to the detected EFI system partition mount-point.</para>
+
+    <para><varname>KERNEL_INSTALL_BOOTLOADER_SPEC=0|1</varname> is set for the plugins to configure whether they should operate on the
+    <ulink url="https://systemd.io/BOOT_LOADER_SPECIFICATION">Boot Loader Specification</ulink> hierarchy.
+    This corresponds to the existence of <filename>$BOOT/<replaceable>MACHINE-ID</replaceable></filename>.</para>
```

Comment:
There are actually two BLS-like layouts: under EFI and under "extended bootloader partition".
I think we might want to be more specific here, and maybe say
KERNEL_INSTALL_LOCATION='efi-bls',
KERNEL_INSTALL_LOCATION='extended-blp-bls' (name TBD :)),
KERNEL_INSTALL_LOCATION='unified-bls' (for single-file 'unified kernel images)
?

And then we could allow the user to explicitly pick the layout by setting the environment variable, and/or add a config file like `/etc/kernel/install.conf` to statically configure this. I think this would make things resilient for people who have some grub installed and get broken boot on every second grub package update.

Reply:
Right, so EFI-BLS would be what 90-kernelentry does, with `$BOOT/loader/entries/MID-KVER[+T].conf` and `$ENTRY_DIR_ABS/{linux,initrds}` (`$BOOT/MID/KVER/{linux,initrds}`), right?

I assume unified-BLS would be something like `$BOOT/MID-KVER.efi`?

extended-blp-BLS would constitute what, exactly?


So if I understand this right, we can do `KERNEL_INSTALL_FLAVOUR=legacy|BLS-EFI|BLS-unified|BLS-EBLP`, and pick between legacy and BLS-EFI as on `$BOOT/MID` presence, if the the config file doesn't have it set to anything, right? This seems most obvious to use and most compatible?

Reply:
See new HEAD. Needs water-treading in the manpage instead of the TODOs, but that requires knowing something about these configs, so

Reply:
Hm, I missed that kernel-install *already* used to `rm -rf $ENTRY_DIR_ABS`, despite it being created by 00-entry-directory.install. And another look through the gist means we can't reasonably go back-to-front on remove, since some hooks (like 20-grubby) exit 77 both for add and remove.

So if the goal is to make it sane again, I don't really see an out that doesn't constitute inlining 00-entry-directory.install into kernel-install? hm

Reply:
OK, makes sense.